cromwell-backend/src/main/scala/cromwell/backend/provider/local/LocalBackend.scala

```diff
@@ -0,0 +1,348 @@
+      .filter(p => p.isDefined).map(p => p.get).toList.distinct
+    val inputVolumes = inputFolder match {
+      case a: List[String] => inputFolder.map(v => dockerVolume.format(v, v)).mkString(" ")
+      case _ => ""
+    }
+    val outputVolume = dockerVolume.format(executionDir, executionDir)
+    log.debug(s"DockerInputVolume: $inputVolumes")
+    log.debug(s"DockerOutputVolume: $outputVolume")
+    val dockerCmd = s"docker run %s %s --rm %s %s" //TODO: make it configurable from file.
+    dockerCmd.format(inputVolumes, outputVolume, image, argv.mkString(" ")).replace("  ", " ")
+  }
+
+  /**
+    * Writes the script file containing the user's command from the WDL as well
+    * as some extra shell code for monitoring jobs
+    */
+  private def writeBashScript(taskCommand: String, executionPath: Path): Unit = {
+    script.write(
+      s"""#!/bin/sh
+          |cd $executionPath
+          |$taskCommand
+          |echo $$? > rc
+          |""".stripMargin)
+  }
+
+  /**
+    * Resolves absolute path for output files. If it is not a file, it will returns same value.
+    * @param output Pair of WdlType and WdlValue
+    * @return WdlValue with absolute path if it is a file.
+    */
+  private def resolveOutputValue(output: (WdlType, Try[WdlValue])): WdlValue = {
+    def getAbsolutePath(file: String): Path = {
+      val absolutePath = Paths.get(executionDir.toString, file)
+      absolutePath.exists match {
+        case true => absolutePath
+        case false => throw new IllegalStateException(s"Output file $file does not exist.")
+      }
+    }
+
+    //TODO: WdlArray[File] is missing here.
+    output._1 match {
+      case lhs if output._2.get.wdlType == WdlFileType | lhs == WdlFileType =>
+        new WdlSingleFile(getAbsolutePath(output._2.get.toWdlString.replace("\"", "").trim).toString)
+      case lhs if lhs == output._2.get.wdlType => output._2.get
+      case lhs => lhs.coerceRawValue(output._2.get).get
+    }
+  }
+
+  /**
+    * Run a command using a bash script.
+    * @return A TaskStatus with the final status of the task.
+    */
+  private def executeTask(): TaskStatus = {
+    def getCmdToExecute(): Seq[String] = dockerImage match {
+      case Some(image) => buildDockerRunCommand(image).split(" ").toSeq
+      case None => argv
+    }
+
+    val process = getCmdToExecute.run(ProcessLogger(stdoutWriter writeWithNewline, stderrTailed writeWithNewline))
+    processAbortFunc = Some(() => process.destroy())
+    subscriptions.filter(subs => subs.eventType.isInstanceOf[ExecutionEvent]).foreach(
+      subs => subs.subscriber ! new TaskStatus(Status.Running, None))
+    val backendCommandString = argv.map(s => "\"" + s + "\"").mkString(" ")
+    logger.debug(s"command: $backendCommandString")
+    val processReturnCode = process.exitValue() // blocks until process finishes
+
+    Vector(stdoutWriter.writer, stderrTailed.writer) foreach {
```

Comment:
The Vector might have come from me way back when. If so, just changed it to a List, my reasoning for preferring Vector turned out to be incorrect.

Also this can be a single line.